File: lib/models/series_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'json_coerce.dart';

part 'series_item.freezed.dart';
part 'series_item.g.dart';

@freezed
abstract class SeriesItem with _$SeriesItem {
  const factory SeriesItem({
    @JsonKey(name: 'series_id') required dynamic seriesId,
    @JsonKey(fromJson: coerceString) @Default('') String name,
    @JsonKey(fromJson: coerceStringOrNull) String? cover,
    @JsonKey(name: 'stream_icon', fromJson: coerceStringOrNull) String? streamIcon,
    @JsonKey(name: 'category_id', fromJson: coerceStringOrNull) String? categoryId,
    @JsonKey(name: 'category_name', fromJson: coerceStringOrNull) String? categoryName,
    @JsonKey(name: 'num_seasons', fromJson: coerceStringOrNull) String? numSeasons,
    @JsonKey(fromJson: coerceStringOrNull) String? rating,
    @JsonKey(fromJson: coerceStringOrNull) String? plot,
    @JsonKey(fromJson: coerceStringOrNull) String? description,
    @JsonKey(fromJson: coerceStringOrNull) String? added,
    @JsonKey(name: 'last_modified', fromJson: coerceStringOrNull) String? lastModified,
  }) = _SeriesItem;

  factory SeriesItem.fromJson(Map<String, dynamic> json) => _$SeriesItemFromJson(json);
}

extension SeriesItemX on SeriesItem {
  String get id => seriesId.toString();
  String get displayIcon => cover ?? streamIcon ?? '';
}

File: tvos/UniStreamTV/UniStreamTV/Models/SeriesItem.swift
import Foundation

/// TV series — mirrors Flutter's `SeriesItem`.
struct SeriesItem: Identifiable, Hashable {
    let seriesId: String
    var name: String
    var cover: String?
    var streamIcon: String?
    var categoryId: String?
    var categoryName: String?
    var numSeasons: String?
    var rating: String?
    var plot: String?
    var description: String?
    var added: String?
    var lastModified: String?

    var id: String { seriesId }
    var displayIcon: String { cover ?? streamIcon ?? "" }

    init(json: [String: Any]) {
        seriesId = "\(json["series_id"] ?? "")"
        name = coerceString(json["name"])
        cover = coerceStringOrNull(json["cover"])
        streamIcon = coerceStringOrNull(json["stream_icon"])
        categoryId = coerceStringOrNull(json["category_id"])
        categoryName = coerceStringOrNull(json["category_name"])
        numSeasons = coerceStringOrNull(json["num_seasons"])
        rating = coerceStringOrNull(json["rating"])
        plot = coerceStringOrNull(json["plot"])
        description = coerceStringOrNull(json["description"])
        added = coerceStringOrNull(json["added"])
        lastModified = coerceStringOrNull(json["last_modified"])
    }
}
